delimited-core/src/main/scala/net/tixxit/delimited/parser/DelimitedParserImpl.scala

```diff
@@ -55,8 +55,20 @@ final case class DelimitedParserImpl(
       case RowDelim(value, Some(alt)) => scala.math.max(value.length, alt.length)
     }
 
+
+    def setSizeHint(ps: ParserState, sizeHint: Int): ParserState =
+      if (ps.sizeHint == sizeHint) ps
+      else
+        ps match {
+          case c@ContinueRow(_, _, _, _, _) => c.copy(sizeHint = sizeHint)
+          case s@SkipRow(_, _, _, _) => s.copy(sizeHint = sizeHint)
+          case p@ParseRow(_, _, _, _) => p.copy(sizeHint = sizeHint)
+        }
+
+    val bldr = Vector.newBuilder[Either[DelimitedError, Row]]
```

Comment:
by using the bldr, each loop also avoids allocating a tuple, in addition to the relatively slow act of calling `:+`. My fast approach used a `List` since `::` is so much faster, but this was even faster, and since it is only local mutability, I think it is fine. What do you think?

Reply:
Yeah, this is great!